widget/widget.go

```diff
@@ -160,23 +158,15 @@ func (w *BaseWidget) setFieldsAndRefresh(f func()) {
 
 	impl := w.getImpl()
 	if impl == nil {
-		w.Refresh()
```

Comment:
I am unsure about this removal.
It means that a widget with no base defined will not refresh...

Reply:
Hmm if a widget has no base defined, then it won't use this, will it? Also `w.Refresh` would not do anything if `impl` is nil, as it is defined as:
```go
// Refresh causes this widget to be redrawn in it's current state
func (w *BaseWidget) Refresh() {
	impl := w.getImpl()
	if impl == nil {
		return
	}
	render := cache.Renderer(impl)
	render.Refresh()
}
```

Reply:
Yes, the combination of the two probably makes this unneeded, sorry.

Reply:
Indeed, I think it should be replaced to just w.Refresh() to avoid repeating code, what do you think?

Reply:
What happens if a widget has overridden `Refresh()`?

Reply:
Oh, I did not take in account that, then it should be simplified to:
```go
	impl := w.getImpl()
	if impl == nil {
		return
	}
	impl.Refresh()
```

Reply:
Yes I think that would probably do it.

Reply:
done in 31dcbd5